adal/src/androidTest/java/com/microsoft/aad/adal/AcquireTokenRequestTest.java

```diff
@@ -554,6 +554,241 @@ public void testVerifyBrokerRedirectUriInvalidSignature()
+            verify(mockedConnection, times(2)).getInputStream();
+            //verify the return
+            assertNotNull(result);
+            assertTrue(result.getAccessToken().equals("I am a new access token"));
+            assertTrue(!result.isExtendedLifeTimeToken());            
+        } catch (final AuthenticationException exception) {
+            fail("Did not expect an exeption");
+        } finally {
+            cacheStore.removeAll();
+        }
+    }
+
+    /**
+     * Test for throwing an exception when ExtendedLifetime is off, the server is down
+     * and AT is expired in the term of expires_in but still valid in ext_expires_in
+     */
+    @SmallTest
+    public void testResiliencyTokenReturn_ExtendedLifetimeOff_expiredAT() throws PackageManager.NameNotFoundException,
+            NoSuchAlgorithmException, OperationCanceledException, IOException, AuthenticatorException,
+            InterruptedException {
+        // make sure AT's expires_in is expired and ext_expires_in is not expired
+        final ITokenCacheStore cacheStore = getTokenCache(-MINUS_MINUITE, false, false, EXTEND_MINUS_MINUTE);
+
+        final FileMockContext mockContext = createMockContext();
+        final AuthenticationContext authContext = new AuthenticationContext(mockContext,
+                VALID_AUTHORITY, false, cacheStore);
+        authContext.setExtendedLifetimeEnabled(false);
+
+        //mock http response
+        final HttpURLConnection mockedConnection = Mockito.mock(HttpURLConnection.class);
+        HttpUrlConnectionFactory.mockedConnection = mockedConnection;
+        Util.prepareMockedUrlConnection(mockedConnection);
+        Mockito.when(mockedConnection.getOutputStream()).thenReturn(Mockito.mock(OutputStream.class));
+        Mockito.when(mockedConnection.getInputStream()).thenReturn(Util.createInputStream(Util.getErrorResponseBody("HTTP_GATEWAY_TIMEOUT")),
+                Util.createInputStream(Util.getErrorResponseBody("HTTP_BAD_GATEWAY")));
+        Mockito.when(mockedConnection.getResponseCode()).thenReturn(HttpURLConnection.HTTP_GATEWAY_TIMEOUT, HttpURLConnection.HTTP_GATEWAY_TIMEOUT);
+
+        try {
+            authContext.acquireTokenSilentSync("resource", "clientid", TEST_USERID);
+            fail("Expect an exeption");
+        } catch (final AuthenticationException exception) {
+            //verify the retry
+            verify(mockedConnection, times(2)).getInputStream();
+            //verify the exception
+            assertNotNull(exception);
+            assertTrue(exception.getCode() == ADALError.AUTH_FAILED_NO_TOKEN);
+        } finally {
+            cacheStore.removeAll();
+        }
+    }
+
+    /**
+     * Test for throwing exception when the request is rejected by server through there is a valid 
+     * stale AT in the cache and the ExtendedLifetime is on.
+     */
+    @SmallTest
+    public void testResiliencyTokenReturn_ExtendedLifetimeOn_noRetry() throws PackageManager.NameNotFoundException,
```

Comment:
>testResiliencyTokenReturn_ExtendedLifetimeOn_noRetry() [](start = 16, length = 54)

Please make sure you have run the checkstyle, the test name you have won't pass checkstyle.